course_contents/2_intro_to_python/notes/03. Comprehensions.md

```diff
@@ -0,0 +1,99 @@
+### The list comprehension
+- the list comprehension allows us to create a list without manually appending objects or running loops
+- the general expression for a list comprehension is:
+
+```py
+new_collection = [element for element in collection] 
+```
+
+- the `element` variable will take the value of each element from the `collection` variable and be appended to the `new_collection`
```

Comment:
Let's say we have an already created list:
`my_list = []`

To that list, we use the `append` method later in the code:
`my_list.append(1)`

A list comprehension creates the list in place, so nothing is appending here, even if it's very similar.

I would explain everything that happens in the list comprehension, then show how you would do it with a normal for loop appending to a list:

![2020-08-05_12-48-13-3c40337cdffdb9b13d4f5101fd7ac38b](https://user-images.githubusercontent.com/69238278/144607874-a6e042c8-15c4-471c-87f4-ccdc23f596f0.JPG)